Module not found: Error: Can't resolve rxjs/*

im using lib in my project 

```
Angular CLI: 6.0.0-rc.2
Node: 8.11.1
OS: win32 x64
Angular: 6.0.0-rc.3
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cdk: 6.0.0-rc.1
@angular/cli: 6.0.0-rc.2
@angular/material: 6.0.0-rc.1
@angular-devkit/architect: 0.5.4
@angular-devkit/build-angular: 0.5.4
@angular-devkit/build-optimizer: 0.5.4
@angular-devkit/core: 0.5.4
@angular-devkit/schematics: 0.5.4
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 6.0.0-rc.2.4
@schematics/angular: 0.5.4
@schematics/update: 0.5.4
typescript: 2.7.2
webpack: 4.5.0
```

it was working on 1.7.4 with 5.x rxjs

today i tried to update to cli rc2 and rxjs rx1 and got this 

```
ERROR in ./node_modules/ngx-infinite-scroll/modules/ngx-infinite-scroll.es5.js
Module not found: Error: Can't resolve 'rxjs/add/observable/fromEvent' in 'C:...node_modules\ngx-infinite-scroll\modules'
ERROR in ./node_modules/ngx-infinite-scroll/modules/ngx-infinite-scroll.es5.js
Module not found: Error: Can't resolve 'rxjs/add/observable/of' in 'C:...node_modules\ngx-infinite-scroll\modules'
ERROR in ./src/app/shared/form-utils/formpart-iban/formpart-iban.component.ts
Module not found: Error: Can't resolve 'rxjs/add/operator/distinctUntilChanged' in 'C:...src\app\shared\form-utils\formpart-iban'
ERROR in ./node_modules/ngx-infinite-scroll/modules/ngx-infinite-scroll.es5.js
Module not found: Error: Can't resolve 'rxjs/add/operator/do' in 'C:...node_modules\ngx-infinite-scroll\modules'
ERROR in ./node_modules/ngx-infinite-scroll/modules/ngx-infinite-scroll.es5.js
Module not found: Error: Can't resolve 'rxjs/add/operator/filter' in 'C:...node_modules\ngx-infinite-scroll\modules'
ERROR in ./node_modules/ngx-infinite-scroll/modules/ngx-infinite-scroll.es5.js
Module not found: Error: Can't resolve 'rxjs/add/operator/map' in 'C:...node_modules\ngx-infinite-scroll\modules'
ERROR in ./node_modules/ngx-infinite-scroll/modules/ngx-infinite-scroll.es5.js
Module not found: Error: Can't resolve 'rxjs/add/operator/mergeMap' in 'C:...node_modules\ngx-infinite-scroll\modules'
ERROR in ./node_modules/ngx-infinite-scroll/modules/ngx-infinite-scroll.es5.js
Module not found: Error: Can't resolve 'rxjs/add/operator/sampleTime' in 'C:...node_modules\ngx-infinite-scroll\modules'
ERROR in ./node_modules/ngx-infinite-scroll/modules/ngx-infinite-scroll.es5.js
Module not found: Error: Can't resolve 'rxjs/observable/of' in 'C:...node_modules\ngx-infinite-scroll\modules'
```

can you maybe add new rxjs syntax sing pipe() thanks

--- a/README.md
+++ b/README.md
@@ -1,67 +1,79 @@
-[![Build Status](https://travis-ci.org/orizens/ngx-infinite-scroll.svg?branch=master)](https://travis-ci.org/orizens/ngx-infinite-scroll) [![Backers on Open Collective](https://opencollective.com/ngx-infinite-scroll/backers/badge.svg)](#backers) [![Sponsors on Open Collective](https://opencollective.com/ngx-infinite-scroll/sponsors/badge.svg)](#sponsors) 
+[![Build Status](https://travis-ci.org/orizens/ngx-infinite-scroll.svg?branch=master)](https://travis-ci.org/orizens/ngx-infinite-scroll) [![Backers on Open Collective](https://opencollective.com/ngx-infinite-scroll/backers/badge.svg)](#backers) [![Sponsors on Open Collective](https://opencollective.com/ngx-infinite-scroll/sponsors/badge.svg)](#sponsors)
+[![npm version](https://badge.fury.io/js/ngx-infinite-scroll.svg)](https://badge.fury.io/js/ngx-infinite-scroll)
 
 # Angular Infinite Scroll
+
 Inspired by [ng-infinite-scroll](https://github.com/sroze/ngInfiniteScroll) directive for angular.
 
 ## Angular Support
+
 **Supports Angular > 4**  
-For Angular version **<= 2.3.1**, you can use ```npm i angular2-infinite-scroll``` (latest version is 0.3.42) - please notice **the angular2-infinite-scroll** package is deprecated
+For Angular version **<= 2.3.1**, you can use `npm i angular2-infinite-scroll` (latest version is 0.3.42) - please notice **the angular2-infinite-scroll** package is deprecated
 
 ## Angular Consulting Services
+
 I'm a Senior Javascript Engineer & A Front End Consultant at [Orizens](http://orizens.com).
-My services include:  
-- consulting to companies and startups on how to approach code in their projects and keep it maintainable.  
-- I provide project bootstrapping and development - while afterwards, I integrate it on site and guide the team on it.  
+My services include:
+
+* consulting to companies and startups on how to approach code in their projects and keep it maintainable.
+* I provide project bootstrapping and development - while afterwards, I integrate it on site and guide the team on it.
 
 [Contact Me Here](http://orizens.com/contact)
 
 ## Installation
+
 ```
 npm install ngx-infinite-scroll --save
 ```
 
 ## Supported API
+
 Currently supported attributes:
-* **infiniteScrollDistance**<_number_> - (optional, default: **2**) - the bottom percentage point of the scroll nob relatively to the infinite-scroll container (i.e, 2 (2 * 10 = 20%) is event is triggered when 80% (100% - 20%) has been scrolled).  
-if container.height is 900px, when the container is scrolled to or past the 720px, it will fire the scrolled event. 
+
+* **infiniteScrollDistance**<_number_> - (optional, default: **2**) - the bottom percentage point of the scroll nob relatively to the infinite-scroll container (i.e, 2 (2 \* 10 = 20%) is event is triggered when 80% (100% - 20%) has been scrolled).  
+  if container.height is 900px, when the container is scrolled to or past the 720px, it will fire the scrolled event.
 * **infiniteScrollUpDistance**<_number_> - (optional, default: **1.5**) - should get a number
-* **infiniteScrollThrottle**<_number_> - (optional, default: **150**) - should get a number of **milliseconds** for throttle. The event will be triggered this many milliseconds after the user *stops* scrolling.
+* **infiniteScrollThrottle**<_number_> - (optional, default: **150**) - should get a number of **milliseconds** for throttle. The event will be triggered this many milliseconds after the user _stops_ scrolling.
 * **infiniteScrollContainer**<_string|HTMLElement_> - (optional, default: null) - should get a html element or css selector for a scrollable element; window or current element will be used if this attribute is empty.
 * **scrolled**<_function_> - this will callback if the distance threshold has been reached on a scroll down.
 * **scrolledUp**<_function_> - (event: InfiniteScrollEvent) - this will callback if the distance threshold has been reached on a scroll up.
 * **scrollWindow**<_boolean_> - (optional, default: **true**) - listens to the window scroll instead of the actual element scroll. this allows to invoke a callback function in the scope of the element while listenning to the window scroll.
 * **immediateCheck**<_boolean_> - (optional, default: **false**) - invokes the handler immediately to check if a scroll event has been already triggred when the page has been loaded (i.e. - when you refresh a page that has been scrolled).
 * **infiniteScrollDisabled**<_boolean_> - (optional, default: **false**) - doesn't invoke the handler if set to true
-* **horizontal**<_boolean_> - (optional, default: false) - sets the scroll to listen for horizontal events  
-* **alwaysCallback**<_boolean_> - (optional, default: false) - instructs the scroller to always trigger events 
-* **fromRoot**<_boolean_> - (optional, default: false) - if **infiniteScrollContainer** is set, this instructs the scroller to query the container selector from the root of the **document** object. 
+* **horizontal**<_boolean_> - (optional, default: false) - sets the scroll to listen for horizontal events
+* **alwaysCallback**<_boolean_> - (optional, default: false) - instructs the scroller to always trigger events
+* **fromRoot**<_boolean_> - (optional, default: false) - if **infiniteScrollContainer** is set, this instructs the scroller to query the container selector from the root of the **document** object.
 
 ## Behavior
+
 By default, the directive listens to the **window scroll** event and invoked the callback.  
 **To trigger the callback when the actual element is scrolled**, these settings should be configured:
+
 * [scrollWindow]="false"
 * set an explict css "height" value to the element
 
 ## DEMO
-- [**Default Scroll** By Window - plunkr](https://plnkr.co/edit/DrEDetYnZkFxR7OWWrxS?p=preview) 
-- [Scroll On a **"Modal"** - plunkr](https://plnkr.co/edit/QnQOwE9SEapwJCCFII3L?p=preview)
+
+* [**Default Scroll** By Window - plunkr](https://plnkr.co/edit/DrEDetYnZkFxR7OWWrxS?p=preview)
+* [Scroll On a **"Modal"** - plunkr](https://plnkr.co/edit/QnQOwE9SEapwJCCFII3L?p=preview)
 
 ## Usage
+
 First, import the InfiniteScrollModule to your module:
 
 ```typescript
-import { NgModule } from '@angular/core';
-import { BrowserModule } from '@angular/platform-browser';
-import { InfiniteScrollModule } from 'ngx-infinite-scroll';
-import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
-import { AppComponent } from './app';
+import { NgModule } from "@angular/core";
+import { BrowserModule } from "@angular/platform-browser";
+import { InfiniteScrollModule } from "ngx-infinite-scroll";
+import { platformBrowserDynamic } from "@angular/platform-browser-dynamic";
+import { AppComponent } from "./app";
 
 @NgModule({
-  imports:[ BrowserModule, InfiniteScrollModule ],
-  declarations: [ AppComponent, ],
-  bootstrap: [ AppComponent ]
+  imports: [BrowserModule, InfiniteScrollModule],
+  declarations: [AppComponent],
+  bootstrap: [AppComponent]
 })
-export class AppModule { }
+export class AppModule {}
 
 platformBrowserDynamic().bootstrapModule(AppModule);
 ```
@@ -69,11 +81,11 @@ platformBrowserDynamic().bootstrapModule(AppModule);
 In this example, the **onScroll** callback will be invoked when the window is scrolled down:
 
 ```typescript
-import { Component } from '@angular/core';
+import { Component } from "@angular/core";
 
 @Component({
-	selector: 'app',
-	template: `
+  selector: "app",
+  template: `
 		<div class="search-results"
 		    infiniteScroll
 		    [infiniteScrollDistance]="2"
@@ -83,25 +95,26 @@ import { Component } from '@angular/core';
 	`
 })
 export class AppComponent {
-	onScroll () {
-	    console.log('scrolled!!')
-	}
+  onScroll() {
+    console.log("scrolled!!");
+  }
 }
 ```
+
 in this example, whenever the "search-results" is scrolled, the callback will be invoked:
 
 ```typescript
-import { Component } from '@angular/core';
+import { Component } from "@angular/core";
 
 @Component({
-	selector: 'app',
-	styles: [
-		`.search-results {
+  selector: "app",
+  styles: [
+    `.search-results {
 			height: 20rem;
 			overflow: scroll;
 		}`
-	],
-	template: `
+  ],
+  template: `
 		<div class="search-results"
 		    infiniteScroll
 		    [infiniteScrollDistance]="2"
@@ -112,22 +125,22 @@ import { Component } from '@angular/core';
 	`
 })
 export class AppComponent {
-	onScroll () {
-	    console.log('scrolled!!')
-	}
+  onScroll() {
+    console.log("scrolled!!");
+  }
 }
 ```
 
 In this example, the **onScrollDown** callback will be invoked when the window is scrolled down and the **onScrollUp** callback will be invoked when the window is scrolled up:
 
 ```typescript
-import { Component } from '@angular/core';
-import { InfiniteScroll } from 'ngx-infinite-scroll';
+import { Component } from "@angular/core";
+import { InfiniteScroll } from "ngx-infinite-scroll";
 
 @Component({
-	selector: 'app',
-	directives: [ InfiniteScroll ],
-	template: `
+  selector: "app",
+  directives: [InfiniteScroll],
+  template: `
 		<div class="search-results"
 		    infiniteScroll
 		    [infiniteScrollDistance]="2"
@@ -139,30 +152,30 @@ import { InfiniteScroll } from 'ngx-infinite-scroll';
 	`
 })
 export class AppComponent {
-	onScrollDown () {
-	    console.log('scrolled down!!')
-	}
+  onScrollDown() {
+    console.log("scrolled down!!");
+  }
 
-	onScrollUp () {
-    	console.log('scrolled up!!')
-    }
+  onScrollUp() {
+    console.log("scrolled up!!");
+  }
 }
 ```
 
 In this example, the **infiniteScrollContainer** attribute is used to point directive to the scrollable container using a css selector. **fromRoot** is used to determine whether the scroll container has to be searched within the whole document (`[fromRoot]="true"`) or just inside the **infiniteScroll** directive (`[fromRoot]="false"`, default option).
 
 ```typescript
-import { Component } from '@angular/core';
+import { Component } from "@angular/core";
 
 @Component({
-	selector: 'app',
-	styles: [
-		`.main-panel {
+  selector: "app",
+  styles: [
+    `.main-panel {
 			height: 100px;
 			overflow-y: scroll;
 		}`
-	],
-	template: `
+  ],
+  template: `
 		<div class="main-panel">
 			<div infiniteScroll
 			    [infiniteScrollDistance]="2"
@@ -175,11 +188,11 @@ import { Component } from '@angular/core';
 	`
 })
 export class AppComponent {
-	selector: string = '.main-panel';
-	
-	onScroll () {
-	    console.log('scrolled!!')
-	}
+  selector: string = ".main-panel";
+
+  onScroll() {
+    console.log("scrolled!!");
+  }
 }
 ```
 
@@ -189,10 +202,8 @@ It is also possible to use **infiniteScrollContainer** without additional variab
 [infiniteScrollContainer]="'.main-panel'"
 ```
 
-
-
-
 # Showcase Examples
+
 * [Echoes Player - Developed with Angular, angular-cli and ngrx](http://orizens.github.io/echoes-player) ([github repo for echoes player](http://github.com/orizens/echoes-player))
 
 ## Contributors
@@ -200,13 +211,11 @@ It is also possible to use **infiniteScrollContainer** without additional variab
 This project exists thanks to all the people who contribute. [[Contribute](CONTRIBUTING.md)].
 <a href="graphs/contributors"><img src="https://opencollective.com/ngx-infinite-scroll/contributors.svg?width=890" /></a>
 
-
 ## Backers
 
 Thank you to all our backers! 🙏 [[Become a backer](https://opencollective.com/ngx-infinite-scroll#backer)]
 
 <a href="https://opencollective.com/ngx-infinite-scroll#backers" target="_blank"><img src="https://opencollective.com/ngx-infinite-scroll/backers.svg?width=890"></a>
-
 
 ## Sponsors
 
@@ -222,5 +231,3 @@ Support this project by becoming a sponsor. Your logo will show up here with a l
 <a href="https://opencollective.com/ngx-infinite-scroll/sponsor/7/website" target="_blank"><img src="https://opencollective.com/ngx-infinite-scroll/sponsor/7/avatar.svg"></a>
 <a href="https://opencollective.com/ngx-infinite-scroll/sponsor/8/website" target="_blank"><img src="https://opencollective.com/ngx-infinite-scroll/sponsor/8/avatar.svg"></a>
 <a href="https://opencollective.com/ngx-infinite-scroll/sponsor/9/website" target="_blank"><img src="https://opencollective.com/ngx-infinite-scroll/sponsor/9/avatar.svg"></a>
-
-

--- a/README.md
+++ b/README.md
@@ -8,6 +8,7 @@ Inspired by [ng-infinite-scroll](https://github.com/sroze/ngInfiniteScroll) dire
 ## Angular Support
 
 **Supports Angular > 4**  
+**Supports rxjs > 5.5**  
 For Angular version **<= 2.3.1**, you can use `npm i angular2-infinite-scroll` (latest version is 0.3.42) - please notice **the angular2-infinite-scroll** package is deprecated
 
 ## Angular Consulting Services

--- a/package.json
+++ b/package.json
@@ -1,7 +1,8 @@
 {
   "name": "ngx-infinite-scroll",
-  "version": "0.8.3",
-  "description": "An infinite scroll directive for Angular compatible with AoT compilation and Tree shaking",
+  "version": "0.8.4",
+  "description":
+    "An infinite scroll directive for Angular compatible with AoT compilation and Tree shaking",
   "main": "./bundles/ngx-infinite-scroll.umd.js",
   "module": "./modules/ngx-infinite-scroll.es5.js",
   "es2015": "./modules/ngx-infinite-scroll.js",
@@ -28,17 +29,14 @@
     "url": "https://github.com/orizens/ngx-infinite-scroll/issues"
   },
   "homepage": "https://github.com/orizens/ngx-infinite-scroll",
-  "keywords": [
-    "angular",
-    "javascript",
-    "typescript"
-  ],
+  "keywords": ["angular", "javascript", "typescript"],
   "license": "MIT",
   "peerDependencies": {
     "@angular/common": ">= 4.0.0",
     "@angular/core": ">= 4.0.0"
   },
   "devDependencies": {
+    "@angular/animations": "^4.0.0",
     "@angular/common": "^4.0.0",
     "@angular/compiler": "^4.0.0",
     "@angular/compiler-cli": "^4.0.0",
@@ -46,9 +44,9 @@
     "@angular/platform-browser": "^4.0.0",
     "@angular/platform-browser-dynamic": "^4.0.0",
     "@angular/platform-server": "^4.0.0",
-    "@angular/animations": "^4.0.0",
     "@types/jasmine": "2.5.46",
     "@types/node": "7.0.10",
+    "chalk": "1.1.3",
     "codelyzer": "3.0.0-beta.4",
     "compodoc": "0.0.41",
     "core-js": "2.4.1",
@@ -61,18 +59,17 @@
     "karma-webpack": "2.0.3",
     "reflect-metadata": "0.1.10",
     "rollup": "0.41.6",
-    "rxjs": "5.5.2",
+    "rxjs": "^5.5.6",
+    "shelljs": "0.7.7",
+    "sorcery": "0.10.0",
     "ts-helpers": "1.1.2",
     "ts-loader": "2.0.3",
     "tslint": "4.5.1",
     "typescript": "^2.1.5",
-    "webpack": "2.3.1",
-    "zone.js": "0.8.5",
-    "shelljs": "0.7.7",
-    "chalk": "1.1.3",
     "uglify-js": "2.8.15",
-    "sorcery": "0.10.0",
-    "yargs": "7.0.2"
+    "webpack": "2.3.1",
+    "yargs": "7.0.2",
+    "zone.js": "0.8.5"
   },
   "dependencies": {
     "opencollective": "^1.0.3"

--- a/src/services/axis-resolver.ts
+++ b/src/services/axis-resolver.ts
@@ -1,11 +1,48 @@
+const VerticalProps = {
+  clientHeight: "clientHeight",
+  offsetHeight: "offsetHeight",
+  scrollHeight: "scrollHeight",
+  pageYOffset: "pageYOffset",
+  offsetTop: "offsetTop",
+  scrollTop: "scrollTop",
+  top: "top"
+};
+
+const HorizontalProps = {
+  clientHeight: "clientWidth",
+  offsetHeight: "offsetWidth",
+  scrollHeight: "scrollWidth",
+  pageYOffset: "pageXOffset",
+  offsetTop: "offsetLeft",
+  scrollTop: "scrollLeft",
+  top: "left"
+};
+
 export class AxisResolver {
+  private propsMap: any;
+
   constructor(private vertical: boolean = true) {
+    this.propsMap = vertical ? VerticalProps : HorizontalProps;
   }
-  clientHeightKey() { return this.vertical ? 'clientHeight' : 'clientWidth'; }
-  offsetHeightKey() { return this.vertical ? 'offsetHeight' : 'offsetWidth'; }
-  scrollHeightKey() { return this.vertical ? 'scrollHeight' : 'scrollWidth'; }
-  pageYOffsetKey() { return this.vertical ? 'pageYOffset' : 'pageXOffset'; }
-  offsetTopKey() { return this.vertical ? 'offsetTop' : 'offsetLeft'; }
-  scrollTopKey() { return this.vertical ? 'scrollTop' : 'scrollLeft'; }
-  topKey() { return this.vertical ? 'top' : 'left'; }
+  clientHeightKey() {
+    return this.propsMap.clientHeight;
+  }
+  offsetHeightKey() {
+    return this.propsMap.offsetHeight;
+  }
+  scrollHeightKey() {
+    return this.propsMap.scrollHeight;
+  }
+  pageYOffsetKey() {
+    return this.propsMap.pageYOffset;
+  }
+  offsetTopKey() {
+    return this.propsMap.offsetTop;
+  }
+  scrollTopKey() {
+    return this.propsMap.scrollTop;
+  }
+  topKey() {
+    return this.propsMap.top;
+  }
 }

--- a/src/services/scroll-register.ts
+++ b/src/services/scroll-register.ts
@@ -1,29 +1,26 @@
-import 'rxjs/add/observable/fromEvent';
-import 'rxjs/add/observable/of';
-import 'rxjs/add/operator/filter';
-import 'rxjs/add/operator/mergeMap';
-import 'rxjs/add/operator/map';
-import 'rxjs/add/operator/do';
-import 'rxjs/add/operator/sampleTime';
+import { ElementRef } from "@angular/core";
+import { Observable } from "rxjs/Observable";
+import { of } from "rxjs/observable/of";
+import { fromEvent } from "rxjs/observable/fromEvent";
+import { map, mergeMap, tap, sampleTime, filter } from "rxjs/operators";
 
-import { ElementRef } from '@angular/core';
-import { Observable } from 'rxjs/Observable';
-import { map } from 'rxjs/operator/map';
-import { of } from 'rxjs/observable/of';
-import { fromEvent } from 'rxjs/observable/fromEvent';
-
-import * as Models from '../models';
-import { AxisResolver } from './axis-resolver';
-import { shouldTriggerEvents, IScrollConfig } from './event-trigger';
-import { resolveContainerElement } from './ngx-ins-utils';
-import { calculatePoints, createResolver } from './position-resolver';
-import * as ScrollResolver from './scroll-resolver';
+import * as Models from "../models";
+import { AxisResolver } from "./axis-resolver";
+import { shouldTriggerEvents, IScrollConfig } from "./event-trigger";
+import { resolveContainerElement } from "./ngx-ins-utils";
+import { calculatePoints, createResolver } from "./position-resolver";
+import * as ScrollResolver from "./scroll-resolver";
 
 export function createScroller(config: Models.IScroller) {
   const { scrollContainer, scrollWindow, element, fromRoot } = config;
   const resolver = createResolver({
     axis: new AxisResolver(!config.horizontal),
-    windowElement: resolveContainerElement(scrollContainer, scrollWindow, element, fromRoot)
+    windowElement: resolveContainerElement(
+      scrollContainer,
+      scrollWindow,
+      element,
+      fromRoot
+    )
   });
   const { totalToScroll: startWithTotal } = calculatePoints(element, resolver);
   const scrollState: Models.IScrollState = {
@@ -43,33 +40,55 @@ export function createScroller(config: Models.IScroller) {
     up: config.upDistance,
     down: config.downDistance
   };
-  return attachScrollEvent(options)
-    .mergeMap((ev: any) => of(calculatePoints(element, resolver)))
-    .map((positionStats: Models.IPositionStats) =>
-      toInfiniteScrollParams(scrollState.lastScrollPosition, positionStats, distance))
-    .do(({ stats, scrollDown }: Models.IScrollParams) =>
+  return attachScrollEvent(options).pipe(
+    mergeMap((ev: any) => of(calculatePoints(element, resolver))),
+    map((positionStats: Models.IPositionStats) =>
+      toInfiniteScrollParams(
+        scrollState.lastScrollPosition,
+        positionStats,
+        distance
+      )
+    ),
+    tap(({ stats, scrollDown }: Models.IScrollParams) =>
       ScrollResolver.updateScrollState(
         scrollState,
         stats.scrolled,
-        stats.totalToScroll,
-      ))
-    .filter(({ fire, scrollDown, stats: { totalToScroll } }: Models.IScrollParams) =>
-      shouldTriggerEvents(
-        config.alwaysCallback, fire, ScrollResolver.isTriggeredScroll(totalToScroll, scrollState, scrollDown))
-    )
-    .do(({ scrollDown, stats: { totalToScroll } }: Models.IScrollParams) => {
-      ScrollResolver.updateTriggeredFlag(totalToScroll, scrollState, true, scrollDown);
-    })
-    .map(toInfiniteScrollAction);
+        stats.totalToScroll
+      )
+    ),
+    filter(
+      ({ fire, scrollDown, stats: { totalToScroll } }: Models.IScrollParams) =>
+        shouldTriggerEvents(
+          config.alwaysCallback,
+          fire,
+          ScrollResolver.isTriggeredScroll(
+            totalToScroll,
+            scrollState,
+            scrollDown
+          )
+        )
+    ),
+    tap(({ scrollDown, stats: { totalToScroll } }: Models.IScrollParams) => {
+      ScrollResolver.updateTriggeredFlag(
+        totalToScroll,
+        scrollState,
+        true,
+        scrollDown
+      );
+    }),
+    map(toInfiniteScrollAction)
+  );
 }
 
-export function attachScrollEvent(options: Models.IScrollRegisterConfig): Observable<{}> {
-  let obs = fromEvent(options.container, 'scroll');
+export function attachScrollEvent(
+  options: Models.IScrollRegisterConfig
+): Observable<{}> {
+  let obs = fromEvent(options.container, "scroll");
   // For an unknown reason calling `sampleTime()` causes trouble for many users, even with `options.throttle = 0`.
   // Let's avoid calling the function unless needed.
   // See https://github.com/orizens/ngx-infinite-scroll/issues/198
   if (options.throttle) {
-    obs = obs.sampleTime(options.throttle);
+    obs = obs.pipe(sampleTime(options.throttle));
   }
   return obs;
 }
@@ -92,11 +111,13 @@ export function toInfiniteScrollParams(
 }
 
 export const InfiniteScrollActions = {
-  DOWN: '[NGX_ISE] DOWN',
-  UP: '[NGX_ISE] UP'
+  DOWN: "[NGX_ISE] DOWN",
+  UP: "[NGX_ISE] UP"
 };
 
-export function toInfiniteScrollAction(response: Models.IScrollParams): Models.IInfiniteScrollAction {
+export function toInfiniteScrollAction(
+  response: Models.IScrollParams
+): Models.IInfiniteScrollAction {
   const { scrollDown, stats: { scrolled: currentScrollPosition } } = response;
   return {
     type: scrollDown ? InfiniteScrollActions.DOWN : InfiniteScrollActions.UP,

--- a/tests/services/position-resolver.spec.ts
+++ b/tests/services/position-resolver.spec.ts
@@ -1,20 +1,23 @@
-import { ElementRef } from '@angular/core';
+import { ElementRef } from "@angular/core";
 
-import { ContainerRef } from '../../src/models';
-import { AxisResolver } from '../../src/services/axis-resolver';
-import { createResolver, createResolverWithContainer } from '../../src/services/position-resolver';
+import { ContainerRef } from "../../src/models";
+import { AxisResolver } from "../../src/services/axis-resolver";
+import {
+  createResolver,
+  createResolverWithContainer
+} from "../../src/services/position-resolver";
 
-describe('Position Resolver', () => {
+describe("Position Resolver", () => {
   let mockedElement: ElementRef;
   let mockedContainer: ElementRef;
   let axis: AxisResolver;
   let mockDom: ContainerRef;
 
   const createMockDom = () => {
-    const container = document.createElement('section');
-    container.setAttribute('style', 'height: 500px; overflow-y: scroll');
-    const el = document.createElement('div');
-    el.setAttribute('style', 'height: 1000px;');
+    const container = document.createElement("section");
+    container.setAttribute("style", "height: 500px; overflow-y: scroll");
+    const el = document.createElement("div");
+    el.setAttribute("style", "height: 1000px;");
     container.appendChild(el);
     mockedElement = new ElementRef(el);
     mockedContainer = new ElementRef(container);
@@ -26,16 +29,16 @@ describe('Position Resolver', () => {
     mockDom = createMockDom();
   });
 
-  describe('Resolver Maker', () => {
-    it('should create an instance of position resolver', () => {
+  describe("Resolver Maker", () => {
+    it("should create an instance of position resolver", () => {
       const actual = createResolver({
         axis,
-        windowElement: mockDom.element,
+        windowElement: mockDom.element
       });
       expect(actual).toBeDefined();
     });
 
-    it('should return the native element if container is not window', () => {
+    it("should return the native element if container is not window", () => {
       mockDom.element.nativeElement = { mocked: true };
       const resolver = createResolverWithContainer(
         {

